src/screens/ExchangeModal.js

```diff
@@ -275,33 +278,33 @@ export default function ExchangeModal({
 
   // Set default gas limit
   useEffect(() => {
-    if (isEmpty(prevGasPrices) && !isEmpty(gasPrices)) {
+    if (isEmpty(prevGasPrices) && !isEmpty(gasFeeParamsBySpeed)) {
       updateTxFee(defaultGasLimit);
     }
-  }, [gasPrices, defaultGasLimit, updateTxFee, prevGasPrices]);
+  }, [gasFeeParamsBySpeed, defaultGasLimit, updateTxFee, prevGasPrices]);
 
   // Update gas limit
   useEffect(() => {
-    if (!isEmpty(gasPrices)) {
+    if (isEmpty(gasFeeParamsBySpeed)) {
       updateGasLimit();
     }
-  }, [gasPrices, updateGasLimit]);
+  }, [gasFeeParamsBySpeed, gasFeesBySpeed, updateGasLimit]);
```

Comment:
is it intentional to include `gasFeesBySpeed` in the dependency list even tho it is not used in the useEffect?